src/app/service/userGroup/userGroup-signal/userGroup-signal.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable, signal } from '@angular/core'
+import { take } from 'rxjs'
+import { Router } from '@angular/router'
+import { UserGroup } from '../../../models/user.model'
+import { UserGroupService } from '../userGroup.service'
+
+export interface UserGroupState {
+  groups: UserGroup[]
+  error: string | null
+}
+@Injectable({
+  providedIn: 'root',
+})
+export class UserGroupSignal {
+  public userGroups = signal<UserGroup[]>([])
+  public selectedUserGroupSignal = signal<UserGroup | null>(null)
+  public errorSignal = signal<string | null>(null)
+
+  constructor(
+    private router: Router,
+    private userGoupService: UserGroupService
+  ) {}
+
+  loadUserGroups(): void {
+    this.userGoupService
+      .getAllUserGroups()
+      .pipe(take(1))
+      .subscribe({
+        next: (userGroups: UserGroup[]) => this.userGroups.set(userGroups),
+        error: (error: Error) =>
+          this.errorSignal.set(error.message || 'Error loading tasks'),
+      })
+  }
+
+  createUserGroup(userGroup: UserGroup): void {
+    this.userGoupService.createUserGroup(userGroup).subscribe({
```

Comment:
unsubscribe, also I would suggest not subscribing in the service as it is hard to manage, but for simple requests that will not require any modifications to data (like fetching some data from the BE, its okay to subscribe inside the service but also nto forget to unsubscribe once the action is finished)

Reply:
same comment for the rest of the methods below (keep in mind that takeUntil wont work here as services dont terminate like components)